pkg/platform/abstract/platform.go

```diff
@@ -337,8 +340,19 @@ func (ap *Platform) ValidateDeleteProjectOptions(deleteProjectOptions *platform.
 		return nuclio.NewErrConflict("Cannot delete the default project")
 	}
 
-	if err := ap.validateProjectIsEmpty(deleteProjectOptions.Meta.Namespace, projectName); err != nil {
-		return errors.Wrap(err, "Cannot delete non-empty project")
+	// ensure project have no sub resources
+	if deleteProjectOptions.Strategy == platform.DeleteProjectStrategyRestrict {
+
+		// validate project is empty (no related resources such as functions, api gateways, etc)
+		if err := ap.validateProjectIsEmpty(deleteProjectOptions.Meta.Namespace, projectName); err != nil {
+			if builtinerrors.Is(err, platform.ErrProjectContainsAPIGateways) ||
+				builtinerrors.Is(err, platform.ErrProjectContainsFunctions) {
+				ap.Logger.DebugWith("Project contains either functions or api gateways",
```

Comment:
I'd suggest starting with "The project" and remove the "either" as they are not really mutually exclusive. The project may contain _both_.
```suggestion
				ap.Logger.DebugWith("The project contains functions or API gateways",
```
To be more future proof and maintainable, you might consider:
```suggestion
				ap.Logger.DebugWith("The project is not empty",
```
Then even if in the future the project contains more entity types other than functions and API gateways, it will still hold true without revising this message.